File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        vector<int> temp;
        int row = matrix.size() , col = matrix[0].size();
        int trow = matrix.size() , tcol = matrix[0].size();
        for(int j = 0 ; j < col ; j++){
            for(int i = row - 1; i >= 0 ; i--){
                temp.push_back(matrix[i][j]);
            }
        }
        int s = 0;
        for(int i = 0; i < row ; i++){
            for(int j = 0 ; j < col ; j++){
                matrix[i][j] = temp[s];
                s++;
            }
        }    
    }
};

File: 0048-rotate-image/0048-rotate-image.java
class Solution {

    public void rotate(int[][] matrix) {
        int col = matrix[0].length ;
        int row = matrix.length ;
        for(int i = 0 ; i < row ; i++){
            for(int j = i+1 ; j < col ; j++){
                int temp = matrix[i][j] ;
                matrix[i][j] = matrix[j][i] ;
                matrix[j][i] = temp ;
            }
        }
        for (int i = 0; i < row; i++) {
    int left = 0, right = matrix[i].length - 1;
    while (left < right) {
        int temp = matrix[i][left];
        matrix[i][left] = matrix[i][right];
        matrix[i][right] = temp;
        left++;
        right--;
    }
}
    }
}